src/internal/observable/bindCallback.ts

```diff
@@ -173,18 +172,19 @@ export function bindCallback(callbackFunc: Function, scheduler?: SchedulerLike):
  * Observable that delivers the same values the callback would deliver.
  * @name bindCallback
  */
-export function bindCallback<T>(
-  callbackFunc: Function,
-  resultSelector?: Function|SchedulerLike,
+export function bindCallback<T, R>(
+  callbackFunc: any,
+  resultSelector?: any,
   scheduler?: SchedulerLike
-): (...args: any[]) => Observable<T> {
+): (...args: any[]) => Observable<T | R> {
```

Comment:
nitpick: If the `R` type parameter is used only in the return type, it serves no real purpose and it should be possible to replace it with `unknown` (in the return type), 'cause that's what it will be anyway. The actual return inference is made in the overload signature; not the implementation sig.